admin_panel/exam_batch/serializer.py

```diff
@@ -32,3 +32,109 @@ class Meta:
 
     def get_student_count(self, obj):
         return obj.exam_students.count()
+
+
+from rest_framework import serializers
+from admin_panel.exam_batch.model import (
+    ExamBatch, ExamBatchSettings, InternetBasedSettings, 
+    PaperBasedSettings, EvaluatorSettings, ModeratorSettings,
+    ScannerSettings, SelectionCriteria
+)
+
+class SelectionCriteriaSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SelectionCriteria
+        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider creating base serializer class</b></div>
<div id="fix">

Consider consolidating the repeated `exclude` fields into a base serializer class since they are identical across all serializer classes
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -35,49 +35,55 @@
 +class BaseExamBatchSerializer(serializers.ModelSerializer):
 +    class Meta:
 +        abstract = True
 +        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']
 +
 +
  class SelectionCriteriaSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = SelectionCriteria
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class ExamBatchSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = ExamBatchSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class InternetBasedSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = InternetBasedSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class PaperBasedSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = PaperBasedSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class EvaluatorSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = EvaluatorSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class ModeratorSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = ModeratorSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']

  class ScannerSettingsSerializer(serializers.ModelSerializer):
 -    class Meta:
 +    class Meta(BaseExamBatchSerializer.Meta):
          model = ScannerSettings
 -        exclude = ['id', 'exam_batch', 'created_at', 'updated_at']
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/10#issuecomment-2499983195>#fd6475</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged